### DISCUSSION TITLE
What about an `Option` type?

### DISCUSSION THREAD

**Discussion by KotlinIsland** [KotlinIsland/basedmypy#179] [Ideas]
Something like [Scala](https://scala-lang.org/api/3.x/scala/Option.html), [rust](https://doc.rust-lang.org/std/option/enum.Option.html) or [Arrow](https://arrow-kt.io/docs/apidocs/arrow-core/arrow.core/-option/)s `Option` type. Basically a wrapped `T | None` that could provide a lot of useful functionality.

TODO: Draw up ideas/concepts about this.

https://github.com/rustedpy/result

**Comment by Zeckie** [comment]
> :)

It might help the discussion to say what this type should do / how it should be used.
